peps/pep-0752.rst

```diff
@@ -0,0 +1,399 @@
+may create a legitimate-appearing package which would execute malicious code at
+runtime. Not only are users more likely to install such packages but doing so
+taints the perception of the entire project.
+
+Namespacing also would drastically reduce the incidence of
+`typosquatting <https://en.wikipedia.org/wiki/Typosquatting>`__
+because typos would have to be in the prefix itself which is likely to be a
+short, well-known identifier like ``aws-``.
+
+Rationale
+=========
+
+Tolerance for Disruption
+------------------------
+
+Other package ecosystems have generally solved this problem by taking one of
+two approaches: either minimizing or maximizing backwards compatibility.
+
+* `NPM <https://www.npmjs.com>`__ has the concept of
+  `scoped packages <https://docs.npmjs.com/about-scopes>`__ which were
+  `introduced`__ primarily to combat there being a dearth of available good
+  package names (whether a real or perceived phenomenon). When a user or
+  organization signs up they are given a scope that matches their name. For
+  example, the
+  `package <https://www.npmjs.com/package/@google-cloud/storage>`__ for using
+  Google Cloud Storage is ``@google-cloud/storage`` where ``@google-cloud/`` is
+  the scope. Regular user accounts (non-organization) may publish `unscoped`__
+  packages for public use.
+  This approach has the lowest amount of backwards compatibility because every
+  installer and tool has to be modified to account for scopes.
+* `NuGet <https://www.nuget.org>`__ has the concept of
+  `package ID prefix reservation`__ which was
+  `introduced`__ primarily to satisfy users wishing to know where a package
+  came from. A package name prefix may be reserved for use by one or more
+  owners. Every reserved package has a special indication
+  `on its page <https://www.nuget.org/packages/Google.Cloud.Storage.V1>`__ to
+  communicate this. After reservation, any upload with a reserved prefix will
+  fail if the user is not an owner of the prefix. Existing packages that have a
+  prefix that is owned may continue to release as usual. This approach has the
+  highest amount of backwards compatibility because only modifications to
+  indices like PyPI are required and installers do not need to change.
+
+__ https://blog.npmjs.org/post/116936804365/solving-npms-hard-problem-naming-packages
+__ https://docs.npmjs.com/package-scope-access-level-and-visibility
+__ https://learn.microsoft.com/en-us/nuget/nuget-org/id-prefix-reservation
+__ https://devblogs.microsoft.com/nuget/Package-identity-and-trust/
+
+This PEP specifies the NuGet approach of authorized reservation across a flat
+namespace for the following reasons:
+
+* Causing churn for the community is a hard blocker.
+* The NPM approach has the potential to cause confusion for users if we allow
+  unscoped names. Our community has chosen to normalize separator characters
+  and so ``@aws/s3`` would likely be confused with ``@aws-s3``.
+
+Granting Namespaces
+-------------------
```

Comment:
It feels like a more appropriate title here relates to generating income rather than "granting namespaces"!

Reply:
@AA-Turner do you view this PEP has an incoming generating tool for the PSF?  I'm not so sure about that for several reasons.  First, organization roll out on PyPI has been delayed mostly IIUC for legal reasons related to the PSF's inability to sell services.  Second, a feature that is *only* available to organizations who can pay for it probably wouldn't go over very well in the community.  Third, there are certainly non-profit and loose organizations that could benefit from namespaces, e.g. everything from PyTorch to Django to others.

Reply:
I have no actual view on this being revenue-generating, the comment was more tongue-in-cheek as the body of this section outlines a problem (understaffing / lack of resource) and posits this process as a seeming solution especially the "*a constant source of funding*" line. Though it is sort-of cart before the horse, and it would be nice if this were more explicit, if it is a desired goal.

More seriously, I do think that the section needs a new name, given this.

Reply:
I modified the text slightly to indicate that it's not exclusively for paid organizations and I changed the section to `Namespace Applications`

Reply:
Cool.  I think for anything income/funding related we will definitely have to get input from the PSF.